### ISSUE TITLE
"Error opening device!"

### ISSUE THREAD

**Issue by gusbemacbe** [sgan81/apfs-fuse#19]
I have installed all the tools as other users hav suggested in other issues and compiled without errors, but there is an error:

```shell
sudo  apfs-fuse /dev/sdc/ /media/ubuntu/
Opening device /dev/sdc/ failed. Not a directory
Error opening device!
```

I have also tried sdc1 and sdc2.

Here's the list of disks:

```
sdc              7.3G                                   
├─sdc2           7.1G                                   
└─sdc1 vfat      200M                                   EFI
```

My APFS disk is an USB Drive.

**Comment by dents** [comment]
Have you tried "sudo  apfs-fuse /dev/sdc /media/ubuntu/" without the trailing slash?

**Comment by gusbemacbe** [comment]
Yes, but it told the media is not empty and asked me to add `nonempty`.

**Comment by dents** [comment]
Wait you are trying to mount the whole disk. Change it to just the partition:

`sudo apfs-fuse /dev/sdc2 /media/ubuntu/`

**Comment by gusbemacbe** [comment]
It also told it isn't empty and asked me to add 'nonempty. Maybe the media `ubuntu`and I've to create and mount a new disk and test it.

**Comment by dents** [comment]
Do you already have something mounted on /media/ubuntu/ ? Like another USB disk? Try making a new directory and mounting it there.

**Comment by gusbemacbe** [comment]
Yes, `/media/ubuntu/` was mounted and full, I had to unmount, and I have just unmounted, but it also told me it is not empty. I will try to create a new directory to mount new media.

**Comment by gusbemacbe** [comment]
I have just created a new directory `/media/osx/` and it still told me it is not empty and asked me to add nonempty. In spite of givng an error, I found `osx` mounted and everything from an AFPS USB drive appeared on this directory at Nautilus. Is it normal or a bug to tell a new media is not empty?
